integrations/aws/main.py

```diff
@@ -69,19 +73,31 @@ async def _handle_global_resource_resync(
 async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     if kind in iter(ResourceKindsWithSpecialHandling):
         return
+
     await update_available_access_credentials()
     is_global = is_global_resource(kind)
-    async for credentials in get_accounts():
-        if is_global:
-            async for batch in _handle_global_resource_resync(kind, credentials):
-                yield batch
-        else:
-            async for session in credentials.create_session_for_each_region():
-                try:
-                    async for batch in resync_cloudcontrol(kind, session):
-                        yield batch
-                except Exception:
-                    continue
+    semaphore = asyncio.Semaphore(MAX_CONCURRENT_TASKS)
+
+    async def handle_account(
+        credentials: AwsCredentials,
+    ) -> ASYNC_GENERATOR_RESYNC_TYPE:
+        """Function to handle fetching resources for a single account."""
+        async with semaphore:
+            if is_global:
+                async for batch in _handle_global_resource_resync(kind, credentials):
+                    yield batch
+            else:
+                async for session in credentials.create_session_for_each_region():
+                    try:
+                        async for batch in resync_cloudcontrol(kind, session):
+                            yield batch
+                    except Exception:
```

Comment:
I know you didn't add this except Exception, but let's remove it. I think this can be really problematic. If you think otherwise, let's talk

Reply:
lets catch the error, and raise at the end if any exception was found, that way we still fetch for all other accounts without failing them.